src/Compatibility/Microsoft.DotNet.PackageValidation/CompatibleFrameworkInPackageValidator.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.ApiCompatibility;
+using NuGet.Client;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.PackageValidation
+{
+    /// <summary>
+    /// Validates that the api surface of the compatible frameworks.
+    /// </summary>
+    public class CompatibleFrameworkInPackageValidator : IPackageValidator
+    {
+        private string _noWarn;
+        private (string, string)[] _ignoredDifferences;
+        
+        public CompatibleFrameworkInPackageValidator(string noWarn, (string, string)[] ignoredDifferences)
+        {
+            _noWarn = noWarn;
+            _ignoredDifferences = ignoredDifferences;
+        }
+
+        public DiagnosticBag<TargetFrameworkApplicabilityDiagnostics> Validate(Package package)
+        {       
+            IEnumerable<ContentItem> compileAssets = package.CompileAssets.OrderByDescending(t => ((NuGetFramework)t.Properties["tfm"]).Version);
+            ManagedCodeConventions conventions = new ManagedCodeConventions(null);
+
+            while (compileAssets.Count() != 1)
+            {
+                ContentItem compileTimeAsset = GetNextItemToRemove(compileAssets);
+                compileAssets = compileAssets.Where(t => t.Path != compileTimeAsset.Path);
```

Comment:
Would it make sense to put all the `compileTimeAssets` in a Queue or something, and then just remove them as you go. Building up a massive chain of `Where( Path != closure.Path )` doesn't seem ideal.